File: examples/python/_2_DirectAccess/subscriber.py
from SharedPubSub import *
from time import sleep

subscriber = Subscriber_atomic_int("DirectAccess","DirectAccessSubscriberPy",False)

while(True):
    ### Example 1 : Wait for notification ###
    value = subscriber.readValue()
    print("SUBSCRIBER PY:",value)
    sleep(1)

File: examples/nodejs/_2_DirectAccess/subscriber.js
const SharedPubSub = require('sharedpubsub');
const sleep = (ms) => new Promise(resolve => setTimeout(resolve, ms));

const subscriber = new SharedPubSub.Subscriber_atomic_int("DirectAccess", "DirectAccessSubscriberJS", true);
var value = 0;

async function loop() {
    while(1){
        /*--------- Example 1 : Read Value  ---------*/
        value = subscriber.readValue();
        console.log(`SUBSCRIBER JS : ${value} readValue()`);
        await sleep(1000);
    }
}

loop().catch(console.error);